mapk_demon/receivers.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import division
+
+from functools import partial
+
+import numpy as np
+import pyro
+from pyro import sample
+from pyro.distributions import Normal, Uniform
+from torch import tensor
+from mapk_demon.transmitters import mapk_signaling
+from pyro.infer.mcmc import MCMC
+from pyro.infer.mcmc.nuts import NUTS
+from pyro.infer import EmpiricalMarginal
+from causal_demon.inference import infer_dist
+import torch
+import matplotlib.pyplot as plt
+
+def infer_mcmc(model):
+    nuts_kernel = NUTS(model, adapt_step_size=True)
+    return MCMC(nuts_kernel, num_samples=500, warmup_steps=300)
+
+
+def g1(a):
+    return a / (a + 1)
+
+
+def g2(a):
+    return a ** 2 / (a ** 2 + a + 1)
+
+
+def f(mu, N):
+    return N * mu ** (0.5) + mu
+
+
+# Parameters
+
+params = {
```

Comment:
See my comment below, you want to sample

```
params = {
    "beta_3k": LogNormal(500/.15, 1),
    "beta_2k": LogNormal(500/.15, 1),
    "beta_k": LogNormal(500/.15, 1)
}
```
The reason for the LogNormal is because you don't want the parameter value to go below 0 in the optimization.